app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,38 @@
+class PostsController < ApplicationController
+  def index
+    @author = User.find(params[:id])
+    @author_posts = Post.where(author_id: @author.id)
```

Comment:
- [ ] Please I suggest you solve N+1 problem when fetching all posts and their comments for a user, in the post#index action user use `@author_posts = @author.author_posts.includes(:comments)` because In Rails, you can use includes method for eager loading. The [includes](https://apidock.com/rails/ActiveRecord/QueryMethods/includes) method specifies relationships to be included in the result set.